File: concurrent/concurrent.hs
module Main where

import Control.Concurrent
import Control.Monad

microInMillis :: Int
microInMillis = 1000

main :: IO ()
main = do

  forkIO $
    forever $ do
      threadDelay $ 1000 * microInMillis
      putStrLn "Hello World 1"

  forkIO $
    forever $ do
      threadDelay $ 700 * microInMillis
      putStrLn "Hello World 2"

  forever $ threadDelay $ 1000 * microInMillis

File: concurrent/concurrent.go
package main

import (
	"fmt"
	"time"
)

func main() {

	go func() {
		for {
			time.Sleep(1000 * time.Millisecond)
			fmt.Println("Hello World 1")
		}
	}()

	go func() {
		for {
			time.Sleep(700 * time.Millisecond)
			fmt.Println("Hello World 2")
		}
	}()

	for {
		time.Sleep(1000 * time.Millisecond)
	}
}
